src/web/js/dashboard/GoogleAPI.js

```diff
@@ -93,31 +110,449 @@ class GoogleAPI {
+      var data = response.result;
+
+      if (classID in data.classList && studentID in data.studentList){
+        var studentIndex = data.classList[classID].students.indexOf(studentID);
+        var classIndex = data.studentList[studentID].classes.indexOf(classID);
+        data.classList[classID].students.splice(studentIndex, 1);
+        data.studentList[studentID].classes.splice(classIndex, 1);
+      }
+      else {
+        //TODO: not sure what's the best way to throw an error here.
+        return undefined;
+      }
+      return this.savePyretData(data);
+    });
+  }
+
+  /** To create, distribute and manage assignments
+  assignment_info{
+    id: int
+    name: string
+    class: int
+    docID: string //id of the template drive file
+    opened: int [] //student id's that have opened
+    submitted: int [] //student id's that have submitted
+  }
+  */
+
+  getAllAssignments = () => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      return data.assignmentList;
+    });
+  }
+
+  getAssignmentsInClass = (classID) => {
+    return this.getClass(classID).then((classInfo) => {
+      var assignmentIDs = classInfo["assignments"];
+      return this.getAllAssignments().then((assignmentInfo) => {
+        var assignmentObjects = [];
+        for (var i = 0; i < assignmentIDs.length; i++){
+          var assignmentObject = assignmentInfo[assignmentIDs[i]];
+          assignmentObjects.push(assignmentObject);
+        }
+        return assignmentObjects;
+      });
+    });
+  }
+
+  /** Assignment realted code begins here */
+
+  // function to create folder
+  createFolder = (parentID, folderName) => {
+    return window.gapi.client.drive.files.create({
+      resource: {
+        'name' : folderName,
+        'parents' : [parentID],
+        'mimeType' : 'application/vnd.google-apps.folder'
+      }
+    });
+  }
+
+// function to copy files
+  copyFile = (parentID, sourceFileID, destinationFileName) => {
+    return window.gapi.client.drive.files.copy({
+      'fileId': sourceFileID,
+      'parents' : [ parentID ],
+      'name': destinationFileName,
+    });
+  }
+
+  // function that initiates the create and distribute assignment process
+  createAndDistributeAssignment = (classID, assignmentName, assignmentFileID) => {
+    return this.getPyretData().then((response) => {
+      var data = response.result;
+      var assignmentInfo = {
+        id: data.nextAssignmentID,
+        name: assignmentName,
+        class: classID,
+        docID: assignmentFileID,
+        opened: [], //list of studentIDs
+        submitted: [], //list of studentIDs
+      };
+      data.assignmentList[data.nextAssignmentID] = assignmentInfo;
+      data.classList[classID].assignments.push(data.nextAssignmentID);
+      data.nextAssignmentID += 1;
+      return this.savePyretData(data).then(() => {
+        return this.getAppFolderID("pyret").then((greatGrandParent) => {
```

Comment:
Forgot to take this string out of the function arguments! See above for Michael's comments on why we don't need this to be parameterized.